openedx/features/course_experience/views/welcome_message.py

```diff
@@ -0,0 +1,49 @@
+"""
+View logic for handling course welcome messages.
+"""
+
+from django.template.loader import render_to_string
+
+from courseware.courses import get_course_info_section_module, get_course_with_access
+from opaque_keys.edx.keys import CourseKey
+from openedx.core.djangoapps.plugin_api.views import EdxFragmentView
+from web_fragments.fragment import Fragment
+
+
+class WelcomeMessageFragmentView(EdxFragmentView):
+    """
+    A fragment that displays a course's welcome message.
+    """
+    def render_to_fragment(self, request, course_id=None, **kwargs):
+        """
+        Renders the welcome message fragment for the specified course.
+
+        Returns: A fragment, or None if there is no welcome message.
+        """
+        course_key = CourseKey.from_string(course_id)
+        course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=True)
```

Comment:
Unfortunately this doesn't work because course updates are stored as XBlocks in the module store. They also aren't available in the course blocks API, so this seems like an area that will need improving if we want to have no modulestore access in the LMS. FYI @nasthagiri.